include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -270,9 +367,9 @@ return_value(bool __res, _Size __size, ::std::true_type)
     return __res ? 0 : __size;
 }
 
-template <typename _Size>
-_Size
-return_value(_Size __res, _Size __size, ::std::false_type)
+template <typename _Size1, typename _Size2>
+auto
+return_value(_Size1 __res, _Size2 __size, ::std::false_type)
```

Comment:
Your overloads of `return_value` has been implemented here not like in other cases: you required only `std::true_type` or `std::false_type` here but in other places we use `_Semantic::value` for example.

Reply:
I don't quite understand what issue you mean here?

Reply:
The comment is not actual any more. I've removed this helper function.